src/me/corriekay/pokegoutil/controllers/AccountController.java

```diff
@@ -186,33 +201,146 @@ public static void logOn() throws Exception {
             UIManager.put("OptionPane.cancelButtonText", "Cancel");
 
             if (cp != null)
-                go = new PokemonGo(cp, http);
-            else
+				try {
+					go = new PokemonGo(cp, http);
+				} catch (LoginFailedException | RemoteServerException e) {	
+					//e.printStackTrace();
+					alertFailedLogin(e.getMessage());
+                    deleteLoginData(LoginType.BOTH);
+                    continue;
+				}
+			else
                 throw new IllegalStateException();
             S_INSTANCE.logged = true;
         }
         S_INSTANCE.go = go;
         initOtherControllers(go);
         S_INSTANCE.mainWindow = new PokemonGoMainWindow(go, S_INSTANCE.console);
         S_INSTANCE.mainWindow.start();
-    }
 
-    private static void initOtherControllers(PokemonGo go) {
-        InventoryController.initialize(go);
-        PokemonBagController.initialize(go);
-    }
+	}
+	
+	public static void logOnPTC(String username, String password) {
+		S_INSTANCE.http = new OkHttpClient();
+		while(!S_INSTANCE.logged) {		
+			//Using PTC, remove Google infos
+			config.delete("login.GoogleAuthToken");
+			try {
+				S_INSTANCE.cp = new PtcCredentialProvider(S_INSTANCE.http, username, password);
+				config.setString("login.PTCUsername", username);
+				if (config.getBool("login.SaveAuth", false) || checkSaveAuth()) {
+					config.setString("login.PTCPassword", password);
+					config.setBool("login.SaveAuth", true);
+				} else {
+					config.delete("login.PTCPassword");
+					config.delete("login.SaveAuth");
+				}
+			} catch(Exception e){
+				alertFailedLogin2();
+				continue;
+			}
+		}
+		
+	}
+	
+	public static void logOnGoogleAuth() {
+		S_INSTANCE.go = null;
+		S_INSTANCE.cp = null;
+		S_INSTANCE.http = new OkHttpClient();
+		while(!S_INSTANCE.logged) {
+			//Using Google, remove PTC infos
+			config.delete("login.PTCUsername");
+			config.delete("login.PTCPassword");
+			String authCode = config.getString("login.GoogleAuthToken", null);
+			boolean refresh = false;
+			if(authCode == null) {
+				//We need to get the auth code, as we do not have it yet.
+				UIManager.put("OptionPane.okButtonText", "Ok");
+				JOptionPane.showMessageDialog(null, "You will need to provide a google authentication key to log in. Press OK to continue.", "Google Auth", JOptionPane.PLAIN_MESSAGE);
+				//We're gonna try to load the page using the users browser 
+				if(Desktop.isDesktopSupported()) {
```

Comment:
Seems like you copied the old code before my fix. You should use our own Desktop class now.
To see how to implement, take a look at the old login functionality.

(And on a sidenote, if that's the new login function, why are you using GUI elements in here? :D )